cmake/legate_helper_functions.cmake

```diff
@@ -0,0 +1,384 @@
+macro(legate_include_rapids)
+  if (NOT _LEGATE_HAS_RAPIDS)
+    if(NOT EXISTS ${CMAKE_BINARY_DIR}/RAPIDS.cmake)
+      file(DOWNLOAD https://raw.githubusercontent.com/rapidsai/rapids-cmake/branch-22.08/RAPIDS.cmake
+           ${CMAKE_BINARY_DIR}/RAPIDS.cmake)
+    endif()
+    include(${CMAKE_BINARY_DIR}/RAPIDS.cmake)
+    include(rapids-cmake)
+    include(rapids-cpm)
+    include(rapids-cuda)
+    include(rapids-export)
+    include(rapids-find)
+    set(_LEGATE_HAS_RAPIDS ON)
+  endif()
+endmacro()
+
+function(legate_default_cpp_install target)
+  set(options)
+  set(one_value_args EXPORT)
+  set(multi_value_args)
+  cmake_parse_arguments(
+    OPT
```

Comment:
I would use something like `LEGATE_OPT` instead of `OPT`. I have seen callers code activate bad code paths because they are using variables with the same name and you get side-effects.